### PR TITLE
Removed api router prefix (if present) when using HRef and URL values…

### PR DESCRIPTION
**Pull request by DavidSubiros**
# What

- Extract API Router Version from API Router URL in config
- In links returned in the response, trim the version prefix if it exists.

### How to review

- Make sure code changes make sense
- Make sure unit tests pass

### Who can review

Anyone

### CODE REVIEW

File: handlers/hierarchy.go [hunk 1]
```diff
@@ -311,7 +312,7 @@ func (f *Filter) Hierarchy(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	p := mapper.CreateHierarchyPage(req, h, d, fil, selVals, allVals, dims, name, req.URL.Path, datasetID, ver.ReleaseDate)
+	p := mapper.CreateHierarchyPage(req, h, d, fil, selVals, allVals, dims, name, req.URL.Path, datasetID, ver.ReleaseDate, "/v1")
```
Comment (CarlHembrough) [inline handlers/hierarchy.go:315]: should be using f.APIRouterVersion?
```suggestion
	p := mapper.CreateHierarchyPage(req, h, d, fil, selVals, allVals, dims, name, req.URL.Path, datasetID, ver.ReleaseDate, f.APIRouterVersion)
```

File: handlers/search.go [hunk 2]
```diff
@@ -103,7 +104,7 @@ func (f *Filter) Search(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	p := mapper.CreateHierarchySearchPage(req, searchRes.Items, d, fil, selVals, dims.Items, allVals, name, req.URL.Path, datasetID, ver.ReleaseDate, req.Referer(), req.URL.Query().Get("q"))
+	p := mapper.CreateHierarchySearchPage(req, searchRes.Items, d, fil, selVals, dims.Items, allVals, name, req.URL.Path, datasetID, ver.ReleaseDate, req.Referer(), req.URL.Query().Get("q"), "/v1")
```
Comment (CarlHembrough) [inline handlers/search.go:107]:
```suggestion
	p := mapper.CreateHierarchySearchPage(req, searchRes.Items, d, fil, selVals, dims.Items, allVals, name, req.URL.Path, datasetID, ver.ReleaseDate, req.Referer(), req.URL.Query().Get("q"), f.APIRouterVersion)
```

**Review by CarlHembrough — CHANGES_REQUESTED** [review]
Looks good, please just check the usage of the hardcoded values.

File: handlers/hierarchy.go (hunk 1 above)
Comment (DavidSubiros) [reply handlers/hierarchy.go:315]: thanks! Good catch! I probably thought it was a test, by mistake.

File: handlers/search.go (hunk 2 above)
Comment (DavidSubiros) [reply handlers/search.go:107]: thanks! Good catch! I probably thought it was a test, by mistake.

**Review by CarlHembrough — APPROVED** [review]

**Pull request merged**